lib/sequelize.js

```diff
@@ -1100,7 +1107,7 @@ class Sequelize {
     if (!autoCallback) return transaction.prepareEnvironment(false).return(transaction);
 
     // autoCallback provided
-    return Sequelize._clsRun(() => {
+    return Sequelize._clsRun(this.queued(() => {
```

Comment:
Interesting, from this it seems that you're only fixing this behavior for CLS, does it really fix it for manual transactions as well?

Reply:
I just don't know, but I will look into it that.

- [ ] Look into dead-lock in manual transactions